### PR TITLE
Making sure the full() primitive extracts default dtype from value argument

### PR DESCRIPTION
**Pull request by hkaiser**
Fixies #1125

### CODE REVIEW

File: tests/unit/python/execution_tree/primitives/numpy_dtype.py [hunk 1]
```diff
@@ -112,6 +112,14 @@ def test_full():
 assert (test_full() == np.full((3, 3), 2, dtype=int)).all()
 
 
+@Phylanx
+def test_full_nodtype():
+    return np.full((3, 3), 2.0)
+
+
+assert (test_full_nodtype() == np.full((3, 3), 2.0)).all()
```
Comment (taless474) [inline tests/unit/python/execution_tree/primitives/numpy_dtype.py]: I think this test only compare values (not dtypes)

**Comment by taless474** [comment]
I am sorry, I was wrong about `full_like`. It gets its dtype from the given array, not the given value.
```python
import numpy as np
from phylanx import Phylanx

@Phylanx
def fx_1(dim):
	return np.ones(dim)
	
@Phylanx
def fx_2(dim, n):
	return np.full(dim, n)
	
@Phylanx
def fx_3(a, n):
	return np.full_like(a, n)
	
n = np.array(2,dtype="int64")
dim = (3,2)
a = np.array([[1,2],[3,4],[5,6]], dtype="float64")


assert(fx_1(dim) == np.ones(dim)).all()
assert(fx_1(dim).dtype == np.ones(dim).dtype)        # float64 and float64

assert(fx_2(dim, n) == np.full(dim, n)).all()
assert(fx_2(dim, n).dtype == np.full(dim, n).dtype)  # int64 and int64

assert(fx_3(a, n) == np.full_like(a, n)).all()
assert(fx_3(a, n).dtype == np.full_like(a, n).dtype) # int64 and float64
```
```pytb
Traceback (most recent call last):
  File "test19.py", line 28, in <module>
    assert(fx_3(a, n).dtype == np.full_like(a, n).dtype) # int64 and float64
AssertionError
```

File: tests/unit/python/execution_tree/primitives/numpy_dtype.py (hunk 1 above)
Comment (hkaiser) [reply tests/unit/python/execution_tree/primitives/numpy_dtype.py]: Fair point, will fix.

**Comment by taless474** [comment]
It works perfectly, thanks.

**Pull request merged**